internal/chezmoi/sourcestate.go

```diff
@@ -2815,3 +2832,11 @@ func allEquivalentDirs(sourceStateEntries []SourceStateEntry) bool {
 	}
 	return true
 }
+
+// isAppleDoubleFile returns true if the file looks like and has the
+// expected signature of an AppleDouble file.
+func isAppleDoubleFile(name string, contents []byte) bool {
+	return strings.HasPrefix(path.Base(name), "._") &&
+		bytes.Equal(appleDoubleMagicCode, contents[0:4]) &&
+		bytes.Equal(appleDoubleVersion, contents[4:8])
```

Comment:
This needs a `len(contents) >= 8` check to avoid panicking. Sorry I just realized this.